### PR TITLE
Add custom message support

### PR DESCRIPTION
**Pull request by js6pak**

### CODE REVIEW

File: src/Impostor.Api/Games/IGame.cs [hunk 1]
```diff
@@ -90,5 +90,24 @@ public interface IGame
...
         /// <returns>A <see cref="ValueTask" /> representing the asynchronous operation.</returns>
         ValueTask SendToAsync(IMessageWriter writer, int id);
+
+        /// <summary>
+        ///     Creates a MessageWriter with GameData header.
+        /// </summary>
+        /// <remarks>Use with caution, recommended only for advanced developers.</remarks>
+        /// <param name="targetNetId">Net id of the InnerNetObject.</param>
+        /// <param name="callId">Rpc id of the message.</param>
+        /// <param name="targetClientId">Client id of GameDataTo's target.</param>
```
Comment (miniduikboot) [inline src/Impostor.Api/Games/IGame.cs:100]: maybe document what the value `-1` does

File: src/Impostor.Api/Utils/IDateTimeProvider.cs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 using System;
 
-namespace Impostor.Api
+namespace Impostor.Api.Utils
```
Comment (miniduikboot) [inline src/Impostor.Api/Utils/IDateTimeProvider.cs:3]: why move classes from the public api? they were stabilized in this location, why do they need to be moved? this is pointless breakage

**Review by miniduikboot — COMMENTED** [review]
looked through it, didn't compile or run the code

File: src/Impostor.Api/Games/IGame.cs (hunk 1 above)
Comment (js6pak) [reply src/Impostor.Api/Games/IGame.cs:100]: Or just make it a nullable int?

File: src/Impostor.Api/Utils/IDateTimeProvider.cs (hunk 2 above)
Comment (js6pak) [reply src/Impostor.Api/Utils/IDateTimeProvider.cs:3]: To keep things clean, it really wasn't supposed to be in root namespace, and no one used it anyway

File: src/Impostor.Api/Games/IGame.cs (hunk 1 above)
Comment (miniduikboot) [reply src/Impostor.Api/Games/IGame.cs:100]: that's probably even better, but document it

**Pull request merged**